week0002/larrypham/swift/Spiral.swift

```diff
@@ -0,0 +1,52 @@
+/**
+ * Given an integer N, generate a square matrix filled with elements from 1 to n*n in spiral order
+ * with n= 4, we have:
+ *	1  2  3  4
+ *	12 13 14  5
+ *	11 16 15  6
+ *	10  9  8  7
+ **/
+import Foundation
+
+class Solution {
+    
+    func generateMatrix(n: Int) -> [[Int]] {
+        var res = Array<[Int]>(repeating: Array<Int>(repeating: 0, count: n), count: n)
+        var k = 1, i = 0
+        while k <= n * n {
+            var j = i
+            while j < n - i {
```

Comment:
Clearly you are going to the right. Please add a comment before each while loop so people can easily understand your logic.